deepsubtree.go

```diff
@@ -59,3 +59,68 @@ func (dsmst *DeepSparseMerkleSubTree) AddBranch(proof SparseMerkleProof, key []b
 
 	return nil
 }
+
+// GetDescend gets the value of a key from the tree by descending it.
+// Use if a key was _not_ previously added with AddBranch, otherwise use Get.
```

Comment:
Why would someone try to get a key that wasn't added with AddBranch? In what circumstances would that succeed?

Reply:
From https://github.com/celestiaorg/smt/pull/39#discussion_r668080136

>  the code wasn't able to get a key that is expected to have a [non-default] value.

☝️  that exact case. When a key in the original SMT has a non-default value, but wasn't added via `AddBranch` to the DSMST, then was queried. This happens when a malicious fraud proof doesn't provide enough pre-state. You don't want to return `defaultValue` in that case because the key doesn't actually have default value; it has a non-default value but that value wasn't proven.